storage/api/Storage.Samples.Tests/BucketFixture.cs

```diff
@@ -0,0 +1,155 @@
+
+        //upload file to BucketName
+        UploadFileSample uploadFileSample = new UploadFileSample();
+        uploadFileSample.UploadFile(BucketNameGeneric, FilePath);
+
+        Collect(FileName);
+    }
+    public void Dispose()
+    {
+        DeleteBucketSample deleteBucketSample = new DeleteBucketSample();
+        DeleteFileSample deleteFileSample = new DeleteFileSample();
+        foreach (var bucket in TempBucketFiles)
+        {
+            foreach (var file in bucket.Value)
+            {
+                try
+                {
+                    deleteFileSample.DeleteFile(bucket.Key, file);
+                }
+                catch (Exception)
+                {
+                    // Do nothing, we delete on a best effort basis.
+                }
+            }
+        }
+
+        foreach (var bucketName in TempBucketNames)
+        {
+            try
+            {
+                deleteBucketSample.DeleteBucket(bucketName);
+                SleepAfterBucketCreateDelete();
+            }
+            catch (Exception)
+            {
+                // Do nothing, we delete on a best effort basis.
+            }
+        }
+    }
+
+    /// <summary>
+    /// Add an object to delete at the end of the test.
+    /// </summary>
+    /// <returns>The objectName.</returns>
+    private string Collect(string bucketName, string objectName)
+    {
+        SortedSet<string> objectNames;
+        if (!TempBucketFiles.TryGetValue(bucketName, out objectNames))
+        {
+            objectNames = TempBucketFiles[bucketName] = new SortedSet<string>();
+        }
+        objectNames.Add(objectName);
+        return objectName;
+    }
+
+    /// <summary>
+    /// Add an object to delete at the end of the test.
+    /// </summary>
+    /// <returns>The objectName.</returns>
+    public string Collect(string objectName) => Collect(BucketNameGeneric, objectName);
+
+    /// <summary>
+    /// Add a object located in a regional bucket to delete
+    /// at the end of the test.
+    /// </summary>
+    /// <returns>The regional objectName.</returns>
+    public string CollectRegionalObject(string objectName) => Collect(BucketNameRegional, objectName);
+
+    /// <summary>
+    /// Bucket creation/deletion is rate-limited. To avoid making the tests flaky, we sleep after each operation.
+    /// </summary>
+    internal void SleepAfterBucketCreateDelete() => Thread.Sleep(2000);
+
+    internal string GetServiceAccountEmail()
+    {
+        var cred = GoogleCredential.GetApplicationDefault().UnderlyingCredential;
+        switch (cred)
+        {
+            case ServiceAccountCredential sac:
+                return sac.Id;
+            // TODO: We may well need to handle ComputeCredential for Kokoro.
+            default:
+                throw new InvalidOperationException($"Unable to retrieve service account email address for credential type {cred.GetType()}");
+        }
+    }
+
+    internal void DeleteAllHmacKeys()
```

Comment:
I don't know where you are calling this. But remember that tests run roughtly at the same time in Windows and Linux, and this method running in one environment could definetely interfere with the other environment.

Reply:
Removed!!